### PR TITLE
Switch the iOS backend URL from the account menu (Debug builds)

### PR DESCRIPTION
**Pull request by claust**
## Why

Debug builds could only talk to whatever `API_BASE_URL` was baked in at build time. Checking the newly deployed home server meant editing `Secrets.xcconfig` and rebuilding — then rebuilding again to get back to local work. Now Debug builds carry both URLs and flip between them at runtime.

## What changed

Tap the profile icon → **Backend** → **Use Local Backend**, with the currently selected URL shown underneath it. The choice persists across launches (`UserDefaults`), defaulting to local.

- `RELEASE_API_BASE_URL` now reaches Debug builds too, as `RemoteAPIBaseURL` in Info.plist (`ios/project.yml`). No new value is needed in `Secrets.xcconfig` — it reuses the one Release builds already require.
- New `BackendSelection` owns the choice, its persistence, and whether there is anything to switch to.
- `AppModel.useLocalBackend(_:)` performs the switch; `APIClient.baseURL` became mutable so the live client is repointed in place.
- `SignInView`'s footer shows the *selected* backend rather than the build default — that is the screen you land back on after a switch.

## Notes for a reviewer

**Release builds are unaffected.** `APIBaseURL` already resolves to `RELEASE_API_BASE_URL` there, and the whole switch is behind `#if DEBUG` — in `BackendSelection`, in the menu, and in the base-URL resolution. A shipped build cannot be pointed at someone's laptop.

**Session handling is the load-bearing part.** The JWT and every in-flight `puzzle_id` were issued by the backend being left behind, so switching clears the session, resets the wizard, and re-mints a token against the new backend from the persisted Google sign-in — silent in practice, but it does drop an in-progress puzzle. That seemed right for a developer-facing switch; say so if you'd rather it warned first.

**Degrades quietly.** With no `RELEASE_API_BASE_URL` in `Secrets.xcconfig` the toggle hides and only the URL label remains, so a fresh checkout following `Secrets.example.xcconfig` doesn't get a toggle that does nothing.

No deployment hostnames are added to the repo — `project.yml` references the variable, and the generated Info.plist stays gitignored.

## Testing

- `make check-ios` clean (swift-format + SwiftLint strict).
- `make ios-test` — 247 tests pass, 0 failures.
- Deployed to a physical iPhone via `make ios-deploy`; the on-device switch itself is still to be exercised end to end.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ios/Pussel/Support/Config.swift [hunk 1]
```diff
@@ -17,6 +17,17 @@ enum Config {
...
+  /// The deployed backend (`RELEASE_API_BASE_URL` in Config/Secrets.xcconfig),
+  /// or nil when the build has no valid value for it. Release builds already
+  /// resolve `apiBaseURL` to this; Debug builds keep it alongside their local
+  /// URL so `BackendSelection` can flip between the two at runtime.
+  static var remoteAPIBaseURL: URL? {
+    guard let url = URL(string: string(for: "RemoteAPIBaseURL")), url.scheme != nil else {
+      return nil
+    }
+    return url
+  }
```
[the remark this answers is not in the document]
Comment (claust) [reply ios/Pussel/Support/Config.swift:45]: Fixed in 5e0bdae. Confirmed the concern is real — `URL(string:)` gives `https://` a scheme and a nil host, as it does `http:/`.

Worth noting it is also the *likely* truncation here rather than a contrived one: the xcconfigs have to write URLs as `https:/$()/host` so `//` does not start a comment, so a value cut off at the `$()` lands exactly on `https://`.

Applied it to `apiBaseURL` too, which had the same gap — and there it was arguably worse, since that path's `fatalError` exists specifically to fail fast on a broken Release config, and a hostless URL was sailing past it. Both now share `Config.backendURL(from:)`, which also requires an http(s) scheme, with tests in `ConfigTests` covering the hostless, unexpanded-`$(VAR)`, empty, and non-http cases.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
